### PR TITLE
Make sure we are sending the same dimensions to each metric

### PR DESCRIPTION
**Pull request by adam-fowler**
Also
- cleaned up metrics update
- Record 404s as NotFound

### CODE REVIEW

File: Sources/Hummingbird/Middleware/MetricsMiddleware.swift [hunk 1]
```diff
@@ -58,24 +58,25 @@ public struct MetricsMiddleware<Context: RequestContext>: RouterMiddleware {
...
-            } else {
-                dimensions = [
+            ).increment()
+            // increment errors
+            Counter(
+                label: "hb.request.errors",
+                dimensions: [
+                    ("http.route", endpointPath),
                     ("http.request.method", request.method.rawValue),
                     ("error.type", errorType),
```
Comment (Joannis) [inline Sources/Hummingbird/Middleware/MetricsMiddleware.swift:77]: Doesn't this need to change too?

File: Sources/Hummingbird/Middleware/MetricsMiddleware.swift [hunk 2]
```diff
@@ -58,24 +58,25 @@ public struct MetricsMiddleware<Context: RequestContext>: RouterMiddleware {
...
-                dimensions = [
+            let endpointPath = context.endpointPath ?? "NotFound"
+            // increment requests
+            Counter(
+                label: "hb.requests",
+                dimensions: [
                     ("http.route", endpointPath),
                     ("http.request.method", request.method.rawValue),
-                    ("error.type", errorType),
+                    ("http.response.status_code", errorType),
```
[the remark this answers is not in the document]
Comment (adam-fowler) [reply Sources/Hummingbird/Middleware/MetricsMiddleware.swift:68]: At the moment this crashes the prometheus metrics collector. You can't send different dimensions to the same label. See #586

File: Sources/Hummingbird/Middleware/MetricsMiddleware.swift (hunk 1 above)
Comment (adam-fowler) [reply Sources/Hummingbird/Middleware/MetricsMiddleware.swift:77]: The dimensions sent to `hb.request.errors` use `error.type`. The dimensions sent to `hb.requests` use dimension `http.response.status_code`

File: Sources/Hummingbird/Middleware/MetricsMiddleware.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (Joannis) [reply Sources/Hummingbird/Middleware/MetricsMiddleware.swift:68]: I've linked #586 to represent this

**Review by Joannis — APPROVED** [review]

**Pull request merged**